src/libs/antares/array/matrix.hxx

```diff
@@ -1422,46 +1250,22 @@ namespace Antares
 	template<class T, class ReadWriteT>
 	template<class PredicateT>
 	bool Matrix<T,ReadWriteT>::internalSaveCSVFile(const AnyString& filename, uint precision,
-		bool addHint, PredicateT& predicate) const
+		bool print_dimensions, PredicateT& predicate) const
 	{
-		using namespace Yuni;
-
-		// Ugly const_cast but it is to preserve a good public API
-		auto& thisNotConst = const_cast<Matrix&>(*this);
+		JIT::just_in_time_manager jit_mgr(jit, filename);
 
-		// Old values
-		// These values are _mandatory_ especially for empty matrices
-		uint oldOpts   = optNone;
-		uint oldWidth  = (width != 0) ? width : 1;
-		uint oldHeight = height;
+		jit_mgr.record_current_jit_state(width, height);
```

Comment:
On pourrait le faire uniquement si le jit est activé ?
Ca éviterait `jit_mgr.clear_matrix(this);` à la suite.

Reply:
pas compris, désolé...

Reply:
OK après discussion sur fonctionnalité du JIT.